include/dts-functions.sh

```diff
@@ -936,21 +773,21 @@ download_bios() {
 }
 
 download_ec() {
-  if [ -v BIOS_LINK_COMM ] && [ ${BIOS_LINK} == ${BIOS_LINK_COMM} ]; then
+  if [ "${BIOS_LINK}" = "${BIOS_LINK_COMM}" ] || [ "${BIOS_LINK}" = "${BIOS_LINK_COMM_CAP}" ]; then
```

Comment:
You could probably simplify whole function if you just used

```sh
download_ec() {
      if [ "${EC_LINK}" = "${EC_LINK_COMM}" ] || [ "${EC_LINK}" = "${EC_LINK_COMM_CAP}" ]; then
        curl -s -L -f "$EC_LINK" -o "$EC_UPDATE_FILE"
        error_check "Cannot access $FW_STORE_URL while downloading binary. Please
          check your internet connection"
        curl -s -L -f "$EC_HASH_LINK" -o $EC_HASH_FILE
        error_check "Cannot access $FW_STORE_URL while downloading signature. Please
          check your internet connection"
        curl -s -L -f "$EC_SIGN_LINK" -o $EC_SIGN_FILE
        error_check "Cannot access $FW_STORE_URL while downloading signature. Please
          check your internet connection"
      else
        curl -s -L -f -u "$USER_DETAILS" -H "$CLOUD_REQUEST" "$EC_LINK" -o $EC_UPDATE_FILE
        error_check "Cannot access $FW_STORE_URL while downloading binary. Please
          check your internet connection"
        curl -s -L -f -u "$USER_DETAILS" -H "$CLOUD_REQUEST" "$EC_HASH_LINK" -o $EC_HASH_FILE
        error_check "Cannot access $FW_STORE_URL while downloading signature. Please
          check your internet connection"
        curl -s -L -f -u "$USER_DETAILS" -H "$CLOUD_REQUEST" "$EC_SIGN_LINK" -o $EC_SIGN_FILE
        error_check "Cannot access $FW_STORE_URL while downloading signature. Please
          check your internet connection"
     fi
}
```

No need to check if `"$HAVE_EC" == "true"` as `download_ec` is already called only if `HAVE_EC==true`.
But if you want you can leave it as it is (resolve this comment in that case)

Reply:
Done here ed6c6089caa0e541f15150b15a79a7e26c6ae365